Bottom: 8bd9b63f4f60de3a1c8c1ce236dfd16da17e2c27
Top:    a2e87aa696218b9da6cd2a3d829df555dde27323
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-12 01:53:58 +0100

PCB+GL: Fix GL headers for Win32


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index b92c68e..835ee44 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -32,14 +32,21 @@
 #include <math.h>
 #include <assert.h>
 
-/* The Linux OpenGL ABI 1.0 spec requires that we define
- * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
- * in order to get prototypes:
- *   http://www.opengl.org/registry/ABI/
- */
-#define GL_GLEXT_PROTOTYPES 1
-#include <GL/gl.h>
-#include <GL/glu.h>
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#   include <windows.h>
+#   include <GL/gl.h>
+#   include <GL/glu.h>
+#else
+    /* The Linux OpenGL ABI 1.0 spec requires that we define
+     * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
+     * in order to get prototypes:
+     *   http://www.opengl.org/registry/ABI/
+     */
+#   define GL_GLEXT_PROTOTYPES 1
+#   include <GL/gl.h>
+#   include <GL/glu.h>
+#endif
 
 #include "action.h"
 #include "crosshair.h"
@@ -58,7 +65,6 @@
 #include <dmalloc.h>
 #endif
 
-
 triangle_buffer buffer;
 float global_depth = 0;
